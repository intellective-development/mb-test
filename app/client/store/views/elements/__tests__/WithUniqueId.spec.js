
import React from 'react';
import uuid from 'uuid';
import withUniqueId from '../WithUniqueId';

jest.mock('uuid');

describe('WithUniqueId', () => {
  it('renders', () => {
    uuid.mockReturnValue('abc123');
    const WrappedComponent = withUniqueId('my_id')(({my_id}) => (
      <div>{my_id}</div>
    ));

    expect(shallow(
      <WrappedComponent />
    )).toMatchSnapshot();
  });
});
