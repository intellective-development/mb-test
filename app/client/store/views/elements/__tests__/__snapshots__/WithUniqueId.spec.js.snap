// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithUniqueId renders 1`] = `
<Component
  my_id="abc123"
/>
`;
